pages/api/events.js

```diff
@@ -15,22 +15,44 @@ export default async function handler(req, res) {
 export async function getEvents() {
   let events = [];
   try {
+    // events = await Profile.aggregate([
+    //   { $project: { username: 1, events: 1, isEnabled: 1 } },
+    //   { $match: { "events.date.start": { $gt: new Date() }, isEnabled: true } },
+    //   { $unwind: "$events" },
+    //   { $match: { "events.date.end": { $gt: new Date() } } },
+    //   { $sort: { "events.date.start": 1 } },
+    // ]).exec();
+
     events = await Profile.aggregate([
       { $project: { username: 1, events: 1, isEnabled: 1 } },
       { $match: { "events.date.start": { $gt: new Date() }, isEnabled: true } },
       { $unwind: "$events" },
       { $match: { "events.date.end": { $gt: new Date() } } },
       { $sort: { "events.date.start": 1 } },
-    ]).exec();
+      {
+        $group: {
+          _id:"$events.url",
+          usernames: {$addToSet: "$username"},
+          isVirtual: {$first:"$events.isVirtual"},
+          color: {$first:"$events.color"},
+          date: {$first:"$events.date"},
+          url: {$first:"$events.url"},
+          name: {$first:"$events.name"},
+          description : {$first:"$events.description"},
+          isEnabled : {$first:"$isEnabled"},
+        }
+      }
+    ]).exec()
+
```

Comment:
returns objects of this structure

```js
{
    _id: 'https://london-2023.devrelcon.dev',
    usernames: [ 'TRohit20' ],
    isVirtual: false,
    color: 'blue',
    date: {
      start: 2023-06-13T00:00:00.000Z,
      end: 2023-06-14T00:00:00.000Z,
      startFmt: 'Tuesday, 13 June 2023 at 05:30:00 GMT+5:30',
      endFmt: 'Wednesday, 14 June 2023 at 05:30:00 GMT+5:30',
      cfpOpen: undefined
    },
    url: 'https://london-2023.devrelcon.dev',
    name: 'DevRelCon London 2023',
    description: 'Back in its home city for the first time since 2019, DevRelCon London is a two day conference bringing together developer relations, education, marketing, and community professionals for sharing, learning, and communing.',
    isEnabled: true
  }
```